py_code.py

```
@@ -0,0 +1,776 @@
+        else:
+            daily_metric_model: StrategyDailyMetricStatModel = \
+                StrategyDailyMetricStatModel.objects.filter(
+                    date=end_date, strategy_id=strategy_id).first()
+            if daily_metric_model is None:
+                daily_metric_model: StrategyDailyMetricStatModel = \
+                    StrategyDailyMetricStatModel()
+                daily_metric_model.strategy_id = strategy_id
+                daily_metric_model.date = end_date
+            daily_metric_models[daily_model_key] = daily_metric_model
+
+        if metric_field_type in _process_mapping:
```

Ruff (0.8.2)

Undefined name `_process_mapping`

(F821)